src/accel/SharedParams.cc

```diff
@@ -347,7 +356,7 @@ void SharedParams::initialize_core(SetupOptions const& options)
     }();
 
     // Construct along-step action
-    {
+    auto along_step = [&params, &options, &imported] {
```

Comment:
I would move the temporary in the lambda and call the lambda inside `insert`
```suggestion
    params.action_reg->insert([&params, &options, &imported] {
```